src/flagpole/evaluation_context.py

```diff
@@ -61,20 +99,27 @@ class ContextBuilder(Generic[T_CONTEXT_DATA]):
     >>> from flagpole import ContextBuilder, Feature
     >>> builder = ContextBuilder().add_context_transformer(lambda _dict: dict(foo="bar"))
     >>> feature = Feature.from_feature_dictionary(name="foo", feature_dictionary=dict(), context=builder)
-    >>> feature.match(dict())
+    >>> feature.match(EvaluationContext(dict()))
     """
 
     context_transformers: list[Callable[[T_CONTEXT_DATA], EvaluationContextDict]]
     exception_handler: Callable[[Exception], Any] | None
+    identity_fields: set[str]
 
     def __init__(self):
         self.context_transformers = []
         self.exception_handler = None
+        self.identity_fields = set()
 
     def add_context_transformer(
-        self, context_transformer: Callable[[T_CONTEXT_DATA], EvaluationContextDict]
+        self,
+        context_transformer: Callable[[T_CONTEXT_DATA], EvaluationContextDict],
+        identity_fields: list[str] | None = None,
     ) -> ContextBuilder[T_CONTEXT_DATA]:
         self.context_transformers.append(context_transformer)
+        if identity_fields is not None:
+            self.identity_fields.update(identity_fields)
```

Comment:
Does `build()` on `ContextBuilder` need to forward `identity_fields` to the Context?

Reply:
yep! Was planning to write tests around this to ensure we properly generate the same Id for different contexts as well.